code/__HELPERS/text.dm

```diff
@@ -840,3 +840,10 @@ GLOBAL_LIST_INIT(binary, list("0","1"))
 			continue
 		out += prob(replaceprob) ? pick(replacementchars) : char
 	return out.Join("")
+
+/proc/num2loadingbar(percent as num, numSquares = 20, reverse = FALSE)
+	var/loadstring = ""
+	for (var/i in 1 to numSquares)
+		var/limit = reverse ? numSquares - percent*numSquares : percent*numSquares
```

Comment:
does the limit need to be redefined every square? you should just be able to have it below loadstring definition for more performance